controllers/AdminController.php

```diff
@@ -301,16 +316,27 @@ public function actionInfo($id)
      */
     public function actionSwitch($id)
     {
-        $user = $this->findModel($id);
+        $old = Yii::$app->user;
+        $session = Yii::$app->session;
+        $key = self::ORIGINAL_USER_SESSION_KEY;
 
-        $old = \Yii::$app->user;
+        if($id == 'original' && $session->has($key)) {
```

Comment:
I agree to dmeroff. This looks cool and also is logical, so i adjusted the PR according to this.